src/Avalonia.Base/Input/AccessKeyHandler.cs

```diff
@@ -224,7 +231,10 @@ protected virtual void OnPreviewKeyUp(object? sender, KeyEventArgs e)
                     {
                         MainMenu.Open();
                     }
-
+                    if (MainMenu is null)
```

Comment:
We may want to also add `|| !MainMenu.IsVisible` here or combine with pattern matching. If it's not visible, then we currently get the same "sticky" behavior as the original issue where alt is released, but the access keys are still underlined.

Reply:
IMO, this should be completely removed from this PR.
Fixing the underlined keys is completely orthogonal to having numbered keys work.
Plus, at least on windows, keeping the underlines is an accessibility setting that we should respect.